shell/platform/fuchsia/flutter/pointer_delegate.cc

```diff
@@ -287,22 +287,46 @@ flutter::PointerData CreateMouseDraft(const fup_MouseEvent& event,
     ptr.buttons = flutter_buttons;
   }
 
-  // Fuchsia currently provides scroll data in "ticks", not physical pixels.
-  // However, Flutter expects scroll data in physical pixels. To compensate for
-  // lack of guidance, we make up a "reasonable amount".
-  // TODO(fxbug.dev/85388): Replace with physical pixel scroll.
+  // TODO(fxbug.dev/103443): Fuchsia previously only provides scroll data in
+  // "ticks", not physical pixels. Keep the tick bases scrolling and remove
+  // after the transistion done.
   const int kScrollOffsetMultiplier = 20;
 
-  if (sample.has_scroll_v()) {
-    ptr.signal_kind = flutter::PointerData::SignalKind::kScroll;
-    double dy = -sample.scroll_v() * kScrollOffsetMultiplier;  // logical amount
-    ptr.scroll_delta_y = dy;  // Not yet physical; adjusted in Platform View.
+  double dy = 0;
+  double dx = 0;
+  bool is_scroll;
+
+  if (sample.has_scroll_v_physical_pixel()) {
+    dy = -sample.scroll_v_physical_pixel();
+    is_scroll = true;
+  } else if (sample.has_scroll_v()) {
+    dy = -sample.scroll_v() *
+         kScrollOffsetMultiplier;  // logical amount, not yet physical; adjusted
+                                   // in Platform View.
+    is_scroll = true;
+  }
+
+  if (sample.has_scroll_h_physical_pixel()) {
+    dx = sample.scroll_h_physical_pixel();
+    is_scroll = true;
+  } else if (sample.has_scroll_h()) {
+    dx = sample.scroll_h() * kScrollOffsetMultiplier;  // logical amount
+    is_scroll = true;
   }
 
-  if (sample.has_scroll_h()) {
+  if (is_scroll) {
     ptr.signal_kind = flutter::PointerData::SignalKind::kScroll;
```

Comment:
This can't be used with `ptr.kind = flutter::PointerData::DeviceKind::kTrackpad`, `pan_x`, `pan_y`, etc. 
To send trackpad gesture data, you need to use `flutter::PointerData::Change::kPanZoomStart`, `flutter::PointerData::Change::kPanZoomUpdate`, `flutter::PointerData::Change::kPanZoomEnd` .
It seems like fuchsia doesn't have yet the understanding of when trackpad gesture starts and stops, just the deltas in higher precision? In that case, it should continue to use `flutter::PointerData::DeviceKind::kTrackpad` and `scroll_delta_x`, `scroll_delta_y`.

Reply:
Done. Yes, it does not have phase for now.